cms/standard_pages/tests/test_models.py

```diff
@@ -0,0 +1,115 @@
+from django.test import TestCase, override_settings
+from wagtail.test.utils import WagtailTestUtils
+
+from cms.standard_pages.tests.factories import IndexPageFactory, InformationPageFactory
+
+
+class IndexPageTestCase(WagtailTestUtils, TestCase):
+    """Test IndexPage model properties and methods."""
+
+    def setUp(self):
+        self.page = IndexPageFactory(
+            title="Test Index Page",
+            description="This is an example",
+            content="This is the main content",
+            featured_items=None,
+            related_links=None,
+        )
+
+        self.page_url = self.page.url
+
+    def test_no_featured_items_displayed_when_no_children_and_no_custom_featured_items_selected(self):
+        """Test that the Featured Items block isn't displayed when the Index Page has no child pages
+        and no custom Featured Items are specified.
+        """
+        self.page.featured_items = None
+
+        response = self.client.get(self.page_url)
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertNotContains(response, "ons-document-list")
+
+    def test_children_displayed_as_featured_items_when_no_custom_featured_items_selected(self):
+        """Test that the children pages of the Index Page are displayed
+        when no custom Featured Items are specified.
+        """
+        child_page = InformationPageFactory(parent=self.page)
+
+        response = self.client.get(self.page_url)
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertContains(response, child_page.title)
+        self.assertContains(response, child_page.url)
+        self.assertContains(response, child_page.summary)
+
+    def test_custom_featured_item_external_page_is_displayed_correctly(self):
+        """Test that the custom featured items are displayed on the page."""
+        featured_item_external_page = {
+            "type": "featured_item",
+            "value": {
+                "title": "Title of the custom featured item",
+                "description": "Description of the custom featured item",
+                "external_url": "external-url.com",
+            },
+        }
+
+        self.page.featured_items = [featured_item_external_page]
+        self.page.save_revision().publish()
+
+        response = self.client.get(self.page_url)
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertContains(response, featured_item_external_page["value"]["title"])
+        self.assertContains(response, featured_item_external_page["value"]["external_url"])
+        self.assertContains(response, featured_item_external_page["value"]["description"])
+
+    def test_custom_featured_item_internal_page_is_displayed_correctly(self):
+        """Test that the custom featured items are displayed on the page."""
+        internal_page = InformationPageFactory(parent=self.page)
```

Comment:
```suggestion
        internal_page = InformationPageFactory()
        child_page = InformationPageFactory(parent=self.page)
```

then also check that the child page data is not there since we have featured takes precedence over child pages.

Reply:
It seems that Faker will initalize those two objects with exactly the same fields. I tried to quickly find a workaround but with no success, so I'm just going to explicitly pass the values to one of the factory constructors.

Reply:
[Test improvements 2.0](https://github.com/ONSdigital/dis-wagtail/pull/78/commits/612a1db5994dced67b02a37db7f3c3518885204b)